content/ember/v3/send-action.md

```diff
@@ -0,0 +1,87 @@
+---
+id: ember-component.send-action
+title: Use closure actions instead of `sendAction`
+until: '4.0.0'
+since: '3.3'
+---
+
+In Ember 1.13 closure actions were introduced as a recommended replacement for `sendAction`.
+With `sendAction` the developer passes the name of of an action, and when `sendAction` is called Ember.js
+would look up that action in the parent context and invoke it if present.
+This had a handful of caveats:
+
+- Since the action is not looked up until it's about to be invoked, it's easier for a typo in the action's name
+to go undetected.
+
+- Using `sendAction` you cannot receive the return value of the invoked action.
+
+Closure actions solve those problems and on top are also more intuitive to use.
+
+```js
+// controllers/index.js
+export default Controller.extend({
+  actions: {
+    sendData(data) {
+      fetch('/endpoint', { body: JSON.stringify(data) });
+    }
+  }
+})
+```
+
+```handlebars
+{{!-- templates/index.hbs --}}
+{{my-component submit="sendData"}}
+```
+
+```js
+// my-component.js
+this.sendAction('submit');
+```
+
+Should be changed to:
+
+```js
+// controllers/index.js
+export default Controller.extend({
+  actions: {
+    sendData(data) {
+      fetch('/endpoint', { body: JSON.stringify(data) });
+    }
+  }
+})
+```
+
+```handlebars
+{{!-- templates/index.hbs --}}
+{{my-component submit=(action "sendData")}}
+```
+
+```js
+// my-component.js
+this.submit();
+```
+
+Note that with this approach the component MUST receive that `submit` property, while with `sendAction` if
+it didn't it would silently do nothing.
+
+If you dont want `submit` to be mandatory, you have to check for the presence of the action before calling it:
```

Comment:
You could also possibly suggest defaulting the value on the prototype (so that you invoke without a guard).

```js
export default Component.extend({
  submit() {} // <- empty function so that `this.submit()` always works
});
```